### DISCUSSION TITLE
Load balancing of gRPC requests from an external client.

### DISCUSSION THREAD

**Discussion by SeminDM** [linkerd/linkerd2#10797] [Help] [answered]
Hello!
I have a project: there are several gRPC services which deployed in OpenShift and gRPC client that deployed outside OpenShift.
I installed Linkerd in OpenShift and service pods have a Linkerd proxy. Load balancing doesn't work for such case.
All articales which I've read describe a case when client is located in OpenShift.
Is it possible to enable lb for external clients?

Thank you!

**Comment by wmorgan** [comment, marked as answer]
Load balancing is a client-side behavior, so the clients will have to be meshed for them to use Linkerd's load balancing directly.  Alternatively, the clients can talk to an ingress controller, which can be meshed, and the ingress can do the load balancing.

**Reply by SeminDM** [reply]
Could you please explain in more details?
I need to create route, ingress and service. Ingress [should be annotated ](https://linkerd.io/2.12/tasks/using-ingress/#haproxy).
Route should terminate tls. What else required for load balancing by ingress and Linkerd?
